src/main/java/me/ryanhamshire/GPFlags/flags/FlagDef_InfiniteArrows.java

```diff
@@ -24,23 +23,28 @@
     public void onProjectileHit(ProjectileHitEvent event) {
         if (event.getEntityType() != EntityType.ARROW && event.getEntityType() != EntityType.SNOWBALL) return;
 
-        Projectile arrow = event.getEntity();
-
-        ProjectileSource source = arrow.getShooter();
-        if (source == null || !(source instanceof Player)) return;
+        ProjectileSource source = event.getEntity().getShooter();
+        if (!(source instanceof Player)) return;
 
         Player player = (Player) source;
         if (player.getGameMode() == GameMode.CREATIVE) return;
 
-        Flag flag = this.GetFlagInstanceAtLocation(arrow.getLocation(), player);
+        Flag flag = this.GetFlagInstanceAtLocation(event.getEntity().getLocation(), player);
```

Comment:
So removing the projectile object, and grabbing the entity twice from the event seems unnecessary. 
This change I feel was unnecessary since it doesn't improve the code in any way.
These changes should be reverted and only things that need to be changed to improve the code and/or fix an issue should be changed.